src/components/form/form.types.ts

```diff
@@ -33,3 +35,67 @@ export interface FormError {
      */
     message: string;
 }
+
+export interface FormComponent<T = any> {
+    /**
+     * The value of the current property
+     */
+    value: T;
+
+    /**
+     * Whether or not the current property is required
+     */
+    required?: boolean;
+
+    /**
+     * Whether or not the current property is readonly
+     */
+    readonly?: boolean;
+
+    /**
+     * Whether or not the current property is disabled
+     */
+    disabled?: boolean;
+
+    /**
+     * The name of the current property
+     */
+    name?: string;
+
+    /**
+     * The title of the current property
+     */
+    title?: string;
```

Comment:
We might have to change this to something else. All these will become props on the component, which means they will become HTML attributes as well. `title` is already a defined HTML attribute and it might add an ugly native tooltip to the element :dizzy_face:

Reply:
Saw that was planning on relabeling to label. :)